NetCasbin/InternalEnforcer.cs

```diff
@@ -17,34 +17,26 @@ public class InternalEnforcer : CoreEnforcer
         /// <returns></returns>
         protected Boolean AddPolicy(String sec, String ptype, List<String> rule)
         {
-            Boolean ruleAdded = model.AddPolicy(sec, ptype, rule);
-            if (!ruleAdded)
-            {
-                return false;
-            } 
-
             if (adapter != null && autoSave)
             {
                 try
                 {
                     adapter.AddPolicy(sec, ptype, rule);
                 }
-                catch (NotImplementedException e)
-                {
-
-                }
-                catch (Exception e)
-                {
-                    throw e;
-                }
-
-                if (watcher != null)
+                catch (NotImplementedException)
                 {
                     // error intentionally ignored
-                    watcher.Update();
                 }
             }
-            return true;
+
+            var ruleAdded = model.AddPolicy(sec, ptype, rule);
+
+            if (ruleAdded)
+            {
+                watcher?.Update();
```

Comment:
Move `watcher?.Update()` to  line 31

Reply:
We should call `watcher?.Update()` before the memory changed include delete?

Reply:
yes.

Reply:
It's better to add `watcher?.Update()` to `try...finally` of adapter.

Reply:
Ok, I will add it to `finally` of adapter.